framework/src/main/java/org/checkerframework/common/accumulation/AccumulationAnnotatedTypeFactory.java

```diff
@@ -663,6 +682,6 @@ protected String convertToPredicate(AnnotationMirror anno) {
      * @return true if anno is a predicate annotation
      */
     protected boolean isPredicate(AnnotationMirror anno) {
-        return predicate != null && AnnotationUtils.areSameByClass(anno, predicate);
+        return predicateName != null && AnnotationUtils.areSameByName(anno, predicateName);
```

Comment:
You should use `AnnotatedTypeFactory#areSameByClass` instead.  It has a cache from Class to name that makes it much faster than AnnotationUtils.areSameByClass.  (You may want to open a new pull request with just that change and discard the other changes in this branch.)

Reply:
@mernst  is going to add comments to `AnnotationUtils#areSameByClass` and related classes explaining that `AnnotatedTypeFactory#areSameByClass` or  `AnnotationUtils.areSameByName` are faster.